setenv.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2013 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package utils
+import (
+	"strings"
+)
+
+// Setenv sets an environment variable entry in the given env slice (as
+// returned by os.Environ or passed in exec.Cmd.Environ) to the given
+// value. The entry must be in the form "x=y" where x is the name of the
+// environment variable and y is its value.
+//
+// If a value isn't already present in the slice, the entry is appended.
+//
+// The new environ slice is returned.
+func Setenv(env []string, entry string) []string {
+	i := strings.Index(entry, "=")
+	if i == -1 {
+		panic("no = in environment entry")
```

Comment:
Anything that would be actively used in production, should *really* not have a panic() in it, since that tends to have the property of crashing all of Juju instead of just having one worker get restarted.